NPE in linked editing when deleting variable

See 
![npe-le](https://user-images.githubusercontent.com/148698/148370486-c729e8eb-f443-4b31-b79d-7e34d9c9f297.gif)

```
[Error - 11:37:31 AM] Request textDocument/linkedEditingRange failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "com.redhat.qute.parser.template.Parameter.setStartValue(int)" because "currentParameter" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:315)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:320)
	at java.base/java.util.concurrent.CompletableFuture.biApply(CompletableFuture.java:1313)
	at java.base/java.util.concurrent.CompletableFuture$BiApply.tryFire(CompletableFuture.java:1280)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.NullPointerException: Cannot invoke "com.redhat.qute.parser.template.Parameter.setStartValue(int)" because "currentParameter" is null
	at com.redhat.qute.parser.parameter.ParameterParser.parse(ParameterParser.java:56)
	at com.redhat.qute.parser.template.Section.parseParameters(Section.java:312)
	at com.redhat.qute.parser.template.Section.getParameters(Section.java:259)
	at com.redhat.qute.parser.template.Section.getParameterAtOffset(Section.java:290)
	at com.redhat.qute.utils.QutePositionUtility.findBestNode(QutePositionUtility.java:112)
	at com.redhat.qute.services.QuteLinkedEditing.findLinkedEditingRanges(QuteLinkedEditing.java:46)
	at com.redhat.qute.services.QuteLanguageService.findLinkedEditingRanges(QuteLanguageService.java:150)
	at com.redhat.qute.ls.template.TemplateFileTextDocumentService.lambda$linkedEditingRange$19(TemplateFileTextDocumentService.java:236)
	at java.base/java.util.concurrent.CompletableFuture.biApply(CompletableFuture.java:1311)
```

--- a/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/parameter/ParameterParser.java
+++ b/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/parameter/ParameterParser.java
@@ -53,8 +53,12 @@ public class ParameterParser {
 				parameters.add(currentParameter);
 				break;
 			case ParameterValue:
-				currentParameter.setStartValue(tokenOffset);
-				currentParameter.setEndValue(tokenEnd);
+				if (currentParameter != null) {
+					// current parameter can be null when parameter name is not defined
+					// ex : {#let =123} (instead of {#let name=123})
+					currentParameter.setStartValue(tokenOffset);
+					currentParameter.setEndValue(tokenEnd);
+				}
 				break;
 			default:
 			}

--- a/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java
+++ b/qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/services/QuteDiagnostics.java
@@ -34,7 +34,6 @@ import com.redhat.qute.commons.JavaMemberInfo;
 import com.redhat.qute.commons.JavaMethodInfo;
 import com.redhat.qute.commons.JavaParameterInfo;
 import com.redhat.qute.commons.ResolvedJavaTypeInfo;
-import com.redhat.qute.parser.expression.MemberPart;
 import com.redhat.qute.parser.expression.MethodPart;
 import com.redhat.qute.parser.expression.NamespacePart;
 import com.redhat.qute.parser.expression.ObjectPart;
@@ -344,6 +343,8 @@ class QuteDiagnostics {
 				}
 			}
 			return resolvedJavaType;
+		} catch (CancellationException e) {
+			throw e;
 		} catch (Exception e) {
 			LOGGER.log(Level.SEVERE, "Error while validating expression '" + expression.getContent() + "' in '"
 					+ template.getUri() + "'.", e);
@@ -598,7 +599,8 @@ class QuteDiagnostics {
 			return null;
 		}
 		String memberType = javaMember.resolveJavaElementType(iterableOfType);
-		return validateJavaTypePart(part, ownerSection, projectUri, diagnostics, resolvingJavaTypeContext, memberType);	}
+		return validateJavaTypePart(part, ownerSection, projectUri, diagnostics, resolvingJavaTypeContext, memberType);
+	}
 
 	/**
 	 * Validate the given method part.

--- a/qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/linkededitingrange/QuteLinkedEditingRangeInForSectionTest.java
+++ b/qute.ls/com.redhat.qute.ls/src/test/java/com/redhat/qute/services/linkededitingrange/QuteLinkedEditingRangeInForSectionTest.java
@@ -60,4 +60,10 @@ public class QuteLinkedEditingRangeInForSectionTest {
 		testLinkedEditingFor(template, null);
 	}
 
+	@Test
+	public void emptyParameterName() throws BadLocationException {
+		String template = "{#let |=3}\r\n" + //
+				"{/let}";
+		testLinkedEditingFor(template, null);
+	}
 }

